### ISSUE TITLE
conflicting documentation for emqx_webhooks and http server

### ISSUE THREAD

**Issue by kiranpradeep** [emqtt/emq-docs-cn#2774]
Kindly find below two screenshot. In http server page, it mention use webhook for simple usecases. But in the breaking changes page (4 vs 5), it mentions that in v5, webhook feature is native and will be referred to as http server.

In addition, the opensoure configuration page [[1](https://docs.emqx.com/en/emqx/v5.8.3/hocon/)] does not show how to use hocon to configure webhook. I like to understand recommendation. For simple usecase, should we use the webhook plugin [[2](https://github.com/emqx/emqx-web-hook)]? If yes how to configure with hocon?

![Screenshot 2024-12-16 at 9 46 39 AM](https://github.com/user-attachments/assets/846bdd37-8b8e-4324-824d-7dd044a847f1)
![Screenshot 2024-12-16 at 9 46 27 AM](https://github.com/user-attachments/assets/b128cf72-c577-48d1-a35b-7308f63ac513)

**Comment by id** [comment]
Indeed, the documentation is not fully correct to say "it is now referred to as HTTP Server bridge". But the rest of the statement stays true: Webhook is now builtin, no plugin is required, and you can use it for simple cases: https://docs.emqx.com/en/emqx/latest/data-integration/webhook.html

**Comment by kiranpradeep** [comment]
Thanks. Could you please also reply to second part of original question.

> If yes how to configure with hocon?

I checked hocon page [[1](https://docs.emqx.com/en/emqx/v5.8.6/hocon/)] and could not find documentation to configure webhook with hocon. The _connector_ appears to be tied to _action_.

**Comment by id** [comment]
It is, in fact, created as an HTTP Server bridge (integration). Dashboard uses specific postfix notation to detect webhooks among other HTTP integrations.
In general, the simplest way to get hocon config is to configure something in dashboard, and then check out `cluster.hocon` in the `data/configs` directory.
Here is an example for 5.8.6. I created a webhook with name "test".

```
connectors {
  http {
    test_WH_D {
      enable = true
      headers {content-type = "application/json"}
      url = "http://localhost:8080"
    }
  }
}
actions {
  http {
    test_WH_D {
      connector = test_WH_D
      enable = true
      parameters {
        headers {content-type = "application/json"}
        method = post
      }
    }
  }
}
rule_engine {
  ignore_sys_message = true
  rules {
    test_WH_D {
      actions = [
        "http:test_WH_D"
      ]
      enable = true
      sql = """~
        SELECT
          *
        FROM
          "#"~"""
    }
  }
}
```

**Comment by kiranpradeep** [comment]
Thanks. The tip to check generated hocon from dashboard is helpful. I will follow same. But wanted to bring to you attention below points.

Now the documentation is confusing to a new user. Documentation clearly mentions webhooks avoid rule engines, but the generated hocon is using rule engines. To the enduser, now it appears more like, internal implementation of webhook is an always true rule engine. So there is no real performance (or readability from a simpler hocon) benefit we can gain from it. Thanks again.

![Image](https://github.com/user-attachments/assets/eaedbd64-d3cf-44d7-8a6f-865cd72ba92a)
